mlflow/pyfunc/model.py

```diff
@@ -391,19 +391,390 @@ def predict_stream(
+            logged_agent_info = mlflow.pyfunc.log_model(
+                artifact_path='agent',
+                python_model=os.path.join(os.getcwd(),"agent"),
+                pip_requirements=[
+                    "mlflow",
+                    "openai",
+                ],
+                # Add serving endpoints, tools, and vector search indexes here
+                resources=[<list resources here>],
+            )
+
+    After logging the model, you can query the model with a single dictionary with the
+    :py:class:`ChatAgentRequest <mlflow.types.agent.ChatAgentRequest>` schema. Under the hood, it
+    will be converted into the python objects expected by your ``predict`` and ``predict_stream``
+    methods.
+
+    .. code-block:: python
+
+        loaded_model = mlflow.pyfunc.load_model(tmp_path)
+        loaded_model.predict(
+            {
+                "messages": [{"role": "user", "content": "What is 10 + 10?"}],
+                "context": {"conversation_id": "123", "user_id": "456"},
+            }
+        )
+
+    To make logging ChatAgent models as easy as possible, MLflow has built in the following
+    features:
+
+    - Automatic Model Signature Inference
+        - You do not need to set a signature when logging a ChatAgent
+        - An input and output signature will be automatically set that adheres to the
+          :py:class:`ChatAgentRequest <mlflow.types.agent.ChatAgentRequest>` and
+          :py:class:`ChatAgentResponse <mlflow.types.agent.ChatAgentResponse>` schemas
+    - Metadata
+        - ``{“task”: “agent/v2/chat”}`` will be automatically appended to any metadata that you may
+          pass in when logging the model
+    - Input Example
+        - Providng an input example is optional, ``mlflow.types.agent.CHAT_AGENT_INPUT_EXAMPLE``
+          will be provided by default
+        - If you do provide an input example, ensure it's a dict with the
+          :py:class:`ChatAgentRequest <mlflow.types.agent.ChatAgentRequest>` schema
+
+        - .. code-block:: python
+
+            input_example = {
+                "messages": [{"role": "user", "content": "What is MLflow?"}],
+                "context": {"conversation_id": "123", "user_id": "456"},
+            }
+
+    **Migrating from ChatModel to ChatAgent**
+
+    To convert an existing ChatModel that takes in
+    :py:class:`ChatMessage <mlflow.types.llm.ChatMessage>` and
+    :py:class:`ChatParams <mlflow.types.llm.ChatParams>` and outputs a
+    :py:class:`ChatCompletionResponse <mlflow.types.llm.ChatCompletionResponse>`, do the following:
+
+    - Redefine your class
+    - Redefine functionality from ``load_context`` into the ``__init__`` method because we removed
+      the ``context`` param from ChatAgent's ``predict`` functionality.
+    - Use ``.model_dump_compat()`` instead of ``.to_dict()`` when converting your model's inputs to
+      dictionaries. Ex. [msg.model_dump_compat() for msg in messages] instead of [msg.to_dict() for
```

Comment:
```suggestion
      dictionaries. Ex. `[msg.model_dump_compat() for msg in messages]` instead of `[msg.to_dict() for msg in messages]`
```

Just to add the code formatting